File: backend/schemas/order.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from decimal import Decimal
from schemas.order_item import OrderItemResponse, OrderLineCreate


class OrderBase(BaseModel):
    customer_id: int
    total_amount: Decimal = Field(default=0, ge=0)
    status: str = "Pending"


class OrderCreate(OrderBase):
    items: list[OrderLineCreate] = []


class OrderResponse(OrderBase):
    id: int
    created_at: datetime
    updated_at: datetime
    customer_name: str | None = None
    customer_email: str | None = None
    items: list[OrderItemResponse] = []

    class Config:
        from_attributes = True


class OrderListResponse(BaseModel):
    items: list[OrderResponse]
    total: int
    page: int
    limit: int
    pages: int


class OrderStatusUpdate(BaseModel):
    status: str

File: frontend/my-app/src/types/order.ts
export type OrderItem = { id: number; order_id: number; product_id: number; product_name?: string | null; quantity: number; price: number };
export type Order = { id: number; customer_id?: number; customer_name?: string | null; customer_email?: string | null; total_amount: number; status: string; created_at?: string; updated_at?: string; items?: OrderItem[] };
